pyogrio/util.py

```diff
@@ -11,21 +11,22 @@ def vsi_path(path: str) -> str:
     # Windows drive letters (e.g. "C:\") confuse `urlparse` as they look like
     # URL schemes
     if sys.platform == "win32" and re.match("^[a-zA-Z]\\:", path):
+        # try to detect archive component without url parsing
+        if path.split("!")[0].endswith(".zip"):
+            return f"zip://{path}"
```

Comment:
Should we actually be prepending `/vsizip/` instead of `zip://`, since this is directly returned and doesn't go through `_construct_vsi_path`?

Reply:
I suppose so, because `zip://` prefixed paths (which should work I think) would result in that: 

```
n [27]: pyogrio.util.vsi_path("zip://C:\layer.shp")
Out[27]: '/vsizip/C:\\layer.shp'
```